src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -529,6 +532,43 @@ public TextChannel getRulesChannel()
         return rulesChannel;
     }
 
+    @Nonnull
+    @Override
+    public RestAction<GuildScheduledEvent> retrieveScheduledEventById(@Nonnull String id)
+    {
+        Checks.isSnowflake(id);
+        return new DeferredRestAction<>(getJDA(), GuildScheduledEvent.class,
+                () -> getScheduledEventById(id),
+                () ->
+                {
+                    Route.CompiledRoute route = Route.Guilds.GET_SCHEDULED_EVENT.compile(getId(), id);
+                    return new RestActionImpl<>(getJDA(), route, (response, request) -> api.getEntityBuilder().createGuildScheduledEvent(this, response.getObject(), getIdLong()));
+                });
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<GuildScheduledEvent> retrieveScheduledEventById(long id)
+    {
+        return retrieveScheduledEventById(String.valueOf(id));
+    }
+
+    @Override
+    @Nonnull
+    public List<GuildScheduledEvent> getScheduledEvents()
```

Comment:
Isn't this default implemented in the interface

Reply:
yes it seems it is